### ISSUE TITLE
Log specified http request headers

### ISSUE THREAD

**Issue by mrkwtz** [ory/oathkeeper#361]
**Is your feature request related to a problem? Please describe.**

Every loadbalancer / reverse-proxy in our http "flow" forwards or creates a request-id via X-Request-ID header. When we troubleshoot any error we throw the request-id in Kibana to get all associated requests. For now oathkeeper doesn't allow us to specify the log format (i.e. the fields). So the logs for the request authentication are "missing" in this case.

**Describe the solution you'd like**

I would like to specify http headers that oathkeeper appends to his log output for every request.

**Describe alternatives you've considered**

-/.

**Additional context**

I hope the description is extensive enough. If not, please let me know and I'll try to elaborate a bit more.

#360 is a duplicate - used the wrong account.

**Comment by aeneasr** [comment]
Yeah, that's definitely valid. I think we've implemented X-Span-ID and similar things in Hydra not too long ago, the PR is: https://github.com/ory/hydra/pull/1685

Would that work here also?

**Comment by DeppLearning** [comment]
Hit the same wall today and failed to workaround the issue using the "header" mutators. Would be nice to see this implemented. I'm using a `X-Correlation-Id` header though.

**Comment by AugPro** [comment]
Hello, I'm facing the same issue (I'd like `trace_id` and `span_id` in logs). I wanted to push a PR for this, mostly copying what had been done for hydra in https://github.com/ory/hydra/pull/1685, but the code has been removed since then. Is there a reason why I should NOT copy what has been done in https://github.com/ory/hydra/pull/1685 ?
